include/nccl_ofi_deque.h

```diff
@@ -46,18 +45,17 @@ struct nccl_ofi_deque_t {
 	 * front. insert_back and insert_front add elements to these respective
 	 * locations.
 	 */
-	nccl_ofi_deque_elem_t head;
+	struct nccl_ofi_deque_elem head;
 	/* Lock for deque operations */
 	pthread_mutex_t lock;
 };
-typedef struct nccl_ofi_deque_t nccl_ofi_deque_t;
 
 /*
  * Initialize deque structure.
  *
  * @return zero on success, non-zero on non-success.
  */
-int nccl_ofi_deque_init(nccl_ofi_deque_t **deque_p);
+extern int nccl_ofi_deque_init(struct nccl_ofi_deque **deque_p);
```

Comment:
I have strong feelings this makes the code less readable.  Please don't do this.

Reply:
I had a ton of confusing build failures ultimately rooted in the fact both of these had the following self-referential typedefs defined:

```
typedef struct nccl_ofi_foo_t nccl_ofi_foo_t;
```

When they meant to define `struct nccl_ofi_foo` and to then do `typedef struct nccl_ofi_foo nccl_ofi_foo_t`.

It's definitely your call to make, but I feel strongly that defining structs in terms of typedefs when not strictly required is a code smell. Selfishly, these useless typedefs also break ctags under emacs for me, which is a painpoint, and defining them in simpler terms makes it work again.

Reply:
strong feelings :).

Reply:
I'll drop this and see if it still compiles. Are you fine with at least fixing the recursive typedefs?!